src/transformStateWithClones.js

```diff
@@ -8,6 +8,24 @@
  */
 function transformStateWithClones(state, actions) {
   // write code here
-}
+  let copyState = { ...state };
+  const arrayStates = [];
+
+  for (const action of actions) {
+    if (action.type === 'clear') {
+      copyState = {};
+    } else if (action.type === 'addProperties') {
+      copyState = { ...copyState, ...action.extraData };
+    } else if (action.type === 'removeProperties') {
+      copyState = { ...copyState };
 
+      for (const key of action.keysToRemove) {
+        delete copyState[key];
+      }
+    }
+    arrayStates.push(copyState);
```

Comment:
Issue: The `copyState` object is directly modified in each iteration, which means all entries in `arrayStates` will reference the same object. To fix this, ensure a new copy of the state is pushed into `arrayStates` after each action is applied. This can be done by using the spread operator again when pushing: `arrayStates.push({ ...copyState });`.